src/components/protocol_handler/src/protocol_handler_impl.cc

```diff
@@ -1188,128 +1242,6 @@ RESULT_CODE ProtocolHandlerImpl::HandleControlMessageEndServiceACK(
-      : connection_key_(connection_key)
-      , protocol_handler_(protocol_handler)
-      , session_observer_(session_observer)
-      , connection_id_(connection_id)
-      , session_id_(session_id)
-      , protocol_version_(protocol_version)
-      , hash_id_(hash_id)
-      , service_type_(service_type)
-      , force_protected_service_(force_protected_service)
-      , full_version_(full_version)
-      , payload_(payload) {}
-
-  bool OnHandshakeDone(
-      const uint32_t connection_key,
-      security_manager::SSLContext::HandshakeResult result) OVERRIDE {
-    if (connection_key != connection_key_) {
-      delete[] payload_;
-      return false;
-    }
-    const bool success =
-        result == security_manager::SSLContext::Handshake_Result_Success;
-    // check current service protection
-    const bool was_service_protection_enabled =
-        session_observer_.GetSSLContext(connection_key_, service_type_) != NULL;
-    if (was_service_protection_enabled) {
-      if (!success) {
-        protocol_handler_->SendStartSessionNAck(
-            connection_id_, session_id_, protocol_version_, service_type_);
-      } else {
-        // Could not be success handshake and not already protected service
-        NOTREACHED();
-      }
-    } else {
-      if (success) {
-        session_observer_.SetProtectionFlag(connection_key_, service_type_);
-      }
-      BsonObject params;
-      if (payload_ != NULL) {
-        params = bson_object_from_bytes(payload_);
-      } else {
-        bson_object_initialize_default(&params);
-      }
-      protocol_handler_->SendStartSessionAck(connection_id_,
-                                             session_id_,
-                                             protocol_version_,
-                                             hash_id_,
-                                             service_type_,
-                                             success,
-                                             full_version_,
-                                             params);
-      bson_object_deinitialize(&params);
-    }
-    delete[] payload_;
-    delete this;
-    return true;
-  }
-
-  void OnCertificateUpdateRequired() OVERRIDE {}
-
-  virtual const std::vector<int>& force_protected_service() const {
-    return force_protected_service_;
-  }
-
- private:
-  const uint32_t connection_key_;
-  ProtocolHandlerImpl* protocol_handler_;
-  SessionObserver& session_observer_;
-
-  const ConnectionID connection_id_;
-  const int32_t session_id_;
-  const uint8_t protocol_version_;
-  const uint32_t hash_id_;
-  const ServiceType service_type_;
-  const std::vector<int> force_protected_service_;
-  ProtocolPacket::ProtocolVersion full_version_;
-  uint8_t* payload_;
-};
-}  // namespace
-#endif  // ENABLE_SECURITY
-
-// Suppress warning for deprecated method used within another deprecated method
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
```

Comment:
This should not be removed, it is necessary because this is deprecated and a deprecated method is used within this function. This pragma is to suppress this warning.

Reply:
Plus, the pragma statement at the end of the function is still present.

Reply:
@jacobkeeler good catch. Restored in https://github.com/smartdevicelink/sdl_core/pull/1971/commits/c95042bdce0bc3d598ce8300e7b17d7d55e9e4b4